apps/sync/js/fxsync-webcrypto/fxsyncwebcrypto.js

```diff
@@ -0,0 +1,279 @@
+'use strict';
+
+/* global StringConversion, KeyDerivation, crypto */
+/* exported FxSyncWebCrypto */
+
+// WebCrypto-based client for Firefox Sync.
+
+var FxSyncWebCrypto = (function fxsync_webcrypto() {
+
+  const HKDF_INFO_STR = 'identity.mozilla.com/picl/v1/oldsync';
+
+  // constructor
+  var FxSyncWebCrypto = function() {
+    // Basic check for presence of WebCrypto
+    if (!crypto || !crypto.subtle) {
+      throw new Error('This environment does not support WebCrypto');
+    }
+
+    this.mainSyncKey = null;
+    this.bulkKeyBundle = null;
+  };
+
+  function importKeyBundle(aesKeyAB, hmacKeyAB) {
+    var pAes = crypto.subtle.importKey('raw', aesKeyAB,
+                                          { name: 'AES-CBC', length: 256 },
+                                          true, [ 'encrypt', 'decrypt' ]
+                                    );
+    var pHmac =  crypto.subtle.importKey('raw', hmacKeyAB,
+                                          { name: 'HMAC', hash: 'SHA-256' },
+                                          true, [ 'sign', 'verify' ]
+                                      );
+    return Promise.all([pAes, pHmac]).then(function(results) {
+      return {
+        aes: results[0],
+        hmac: results[1]
+      };
+    });
+  }
+
+  FxSyncWebCrypto.prototype._importKb = function(kBByteArray) {
```

Comment:
Do we need to expose `_importKb` and other private functions on the object prototype?